### PR TITLE
Retry on google drive API error

### PR DESCRIPTION
**Pull request by gseva**
This PR adds a wrapper for each drive API call, that retries the call in case of unexpected errors

### CODE REVIEW

File: laika/reports.py [hunk 1]
```diff
@@ -1095,7 +1095,36 @@ def create_drive(profile, grant):
...
+                    status = e.args[0].resp.status
+                else:
+                    status = e.resp.status
+                if timeout < limit and (status >= 500 or status in self.retry_status_codes):
+                    log = 'An error occured executing %s: %s. Waiting %d seconds.'
+                    logging.info(log, method, e, timeout)
+                    time.sleep(timeout)
+                    timeout = timeout * 2
+                else:
+                    raise
```
Comment (PalmaLeandro) [inline laika/reports.py]: If API raises an error with below 500 and not in `retry_status_codes` I think you should inform it by raising the catched exception.
```suggestion
                    raise e
```

File: docs/results.rst [hunk 2]
```diff
@@ -108,6 +108,12 @@ configurations:
    Must have access to specified folder.
 -  mime_type: Media type of the file to be uploaded. If none is specified
    it will take the type of the filename extension.
+-  start_timeout, max_timeout, retry_status_codes: drive API calls sometimes
+   fail with 500 errors. To work around this behaviour, in case of error the
+   call is retried after waiting *start_timeout* (2 by default) seconds,
+   doubling the waiting time after each error until reaching *max_timeout* (300
```
Comment (PalmaLeandro) [inline docs/results.rst]: I would be nice to mention what happens when it reaches `max_timeout`. An exception or what?

File: laika/reports.py (hunk 1 above)
Comment (gseva) [reply laika/reports.py]: Yeah, it's the default action of `raise`, to reraise the catched exception (so there is no difference between the two in this case).

File: docs/results.rst (hunk 2 above)
Comment (gseva) [reply docs/results.rst]: Done 👍

**Pull request merged**